apps/server/src/modules/oauth/service/hydra.service.spec.ts

```diff
@@ -0,0 +1,248 @@
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { HydraSsoService } from '@src/modules/oauth/service/hydra.service';
+import { LtiToolRepo } from '@shared/repo';
+import { HttpService } from '@nestjs/axios';
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+import { OauthConfig } from '@shared/domain';
+import { of } from 'rxjs';
+import { LtiToolDO } from '@shared/domain/domainobject/ltitool.do';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { InternalServerErrorException } from '@nestjs/common';
+import { CookiesDto } from '@src/modules/oauth/service/dto/cookies.dto';
+import { AuthorizationParams } from '@src/modules/oauth/controller/dto';
+import { HydraRedirectDto } from '@src/modules/oauth/service/dto/hydra.redirect.dto';
+
+class HydraOauthSsoSpec extends HydraSsoService {
+	public processCookiesSpec(setCookies: string[], cookie: CookiesDto): CookiesDto {
+		return super.processCookies(setCookies, cookie);
+	}
+}
+
+jest.mock('nanoid', () => {
+	return {
+		nanoid: jest.fn(() => 'mockNanoId'),
+	};
+});
+
+const createAxiosResponse = <T = unknown>(data: T): AxiosResponse<T> => ({
+	data,
+	status: 0,
+	statusText: '',
+	headers: {},
+	config: {},
+});
+
+describe('HydraService', () => {
+	let module: TestingModule;
+	let service: HydraOauthSsoSpec;
+
+	let ltiToolRepo: DeepMocked<LtiToolRepo>;
+	let httpService: DeepMocked<HttpService>;
+
+	const responseData = {};
+
+	const hydraUri = 'http://hydra.mock';
+	const scopes = 'openid uuid';
+	const apiHost = 'localhost';
+
+	const oauthConfig: OauthConfig = new OauthConfig({
+		clientId: '12345',
+		clientSecret: 'mocksecret',
+		tokenEndpoint: `${hydraUri}/oauth2/token`,
+		grantType: 'authorization_code',
+		redirectUri: `${apiHost}/api/v3/sso/hydra/12345`,
+		scope: scopes,
+		responseType: 'code',
+		authEndpoint: `${hydraUri}/oauth2/auth`,
+		provider: 'hydra',
+		logoutEndpoint: `${hydraUri}/oauth2/sessions/logout`,
+		issuer: `${hydraUri}/`,
+		jwksEndpoint: `${hydraUri}/.well-known/jwks.json`,
+	});
+
+	beforeAll(async () => {
+		jest.spyOn(Configuration, 'get').mockImplementation((key: string): unknown => {
+			switch (key) {
+				case 'HYDRA_PUBLIC_URI':
+					return hydraUri;
+				case 'API_HOST':
+					return apiHost;
+				case 'HOST':
+					return apiHost;
+				case 'NEXTCLOUD_SCOPES':
```

Comment:
HOST and API_HOST both get mapped to apiHost, that seems weird. Pls check again if that is correct. If it is not, you should also adjust the tests, so that such mistakes would be caught by the test ;)